packages/flutter/lib/src/material/tooltip_theme.dart

```diff
@@ -0,0 +1,227 @@
+
+  /// They height of the tooltip's [child].
+  ///
+  /// If the [child] is null, then this is the intrinsic height.
+  final double height;
+
+  /// The amount of space by which to inset the child.
+  ///
+  /// Defaults to 16.0 logical pixels in each direction.
+  final EdgeInsetsGeometry padding;
+
+  /// The vertical gap between the widget and the displayed tooltip.
+  final double verticalOffset;
+
+  /// Whether the tooltip defaults to being displayed below the widget.
+  ///
+  /// Defaults to true. If there is insufficient space to display the tooltip in
+  /// the preferred direction, the tooltip will be displayed in the opposite
+  /// direction.
+  final bool preferBelow;
+
+  /// Whether the tooltip's [message] should be excluded from the semantics
+  /// tree.
+  final bool excludeFromSemantics;
+
+  /// Specifies the tooltip's shape and background color.
+  ///
+  /// If not specified, defaults to a rounded rectangle with a border radius of
+  /// 4.0, and a color derived from the [ThemeData.textTheme] if the
+  /// [ThemeData.brightness] is dark, and [ThemeData.primaryTextTheme] if not.
+  final Decoration decoration;
+
+  /// The style to use for the message of the tooltip.
+  ///
+  /// If null, the message's [TextStyle] will be determined based on
+  /// [ThemeData]. If [ThemeData.brightness] is set to [Brightness.dark],
+  /// [ThemeData.textTheme.body1] will be merged with
+  /// [ThemeData.typography.white]. Otherwise, if [ThemeData.brightness] is set
+  /// to [Brightness.light], [ThemeData.primaryTextTheme.body1] will be merged
+  /// with [ThemeData.typography.white].
+  final TextStyle textStyle;
+
+  /// The amount of time that a pointer must hover over the widget before it
+  /// will show a tooltip.
+  ///
+  /// Defaults to 0 milliseconds (tooltips show immediately upon hover).
+  final Duration waitDuration;
+
+  /// The amount of time that the tooltip will be shown once it has appeared.
+  ///
+  /// Defaults to 1.5 seconds.
+  final Duration showDuration;
+
+  /// Creates a copy of this object but with the given fields replaced with the
+  /// new values.
+  TooltipThemeData copyWith({
+    double height,
+    EdgeInsetsGeometry padding,
+    double verticalOffset,
+    bool preferBelow,
+    bool excludeFromSemantics,
+    Decoration decoration,
+    TextStyle textStyle,
+    Duration waitDuration,
+    Duration showDuration,
+  }) {
+    return TooltipThemeData(
+      height: height ?? this.height,
+      padding: padding ?? this.padding,
+      verticalOffset: verticalOffset ?? this.verticalOffset,
+      preferBelow: preferBelow ?? this.preferBelow,
+      excludeFromSemantics: excludeFromSemantics ?? this.excludeFromSemantics,
+      decoration: decoration ?? this.decoration,
+      textStyle: textStyle ?? this.textStyle,
+      waitDuration: waitDuration ?? this.waitDuration,
+      showDuration: showDuration ?? this.showDuration,
+    );
+  }
+
+  /// Linearly interpolate between two tooltip themes.
```

Comment:
If we're going to prohibit changing preferBelow and excludeFromSemantics, we should document that. It might be better to use this kind of "interpolation" instead:
```
      preferBelow: t < 0.5 ? a.preferBelow: b.preferBelow
```

Ideally this stuff wouldn't change but it doesn't seem worth creating a corner case for the odd case where it does.

Reply:
I'll use that interpolation style instead. It seems to be better than just blocking them outright.